### PR TITLE
fix: gitea config url for avatar

### PR DESCRIPTION
**Pull request by jolheiser**
Rather than getting avatar URL from gitea.com, use the API for the specified Gitea instance in the config.

The second commit is `gofumpt`, let me know if you'd like me to remove it as it's just formatting and unrelated to this PR, it just runs on save on my machine so I left it included as a separate commit.

Non-formatting change is simply https://github.com/thomiceli/opengist/commit/a111a1295544426e4b5f6d3574932dffde739376

### CODE REVIEW

**Comment by thomiceli** [comment]
I have missed that, thanks. But i think should be not used anymore since it can cause problems like #35, maybe simply use `urlJoin()`

File: internal/web/auth.go [hunk 1]
```diff
@@ -351,7 +352,7 @@ func getAvatarUrlFromProvider(provider string, identifier string) string {
 	case "github":
 		return "https://avatars.githubusercontent.com/u/" + identifier + "?v=4"
 	case "gitea":
-		resp, err := http.Get("https://gitea.com/api/v1/users/" + identifier)
+		resp, err := http.Get(trimGiteaUrl() + "/api/v1/users/" + identifier)
```
Comment (thomiceli) [inline internal/web/auth.go:355]: Something like `http.Get(urlJoin(config.C.GiteaUrl, "/api/v1/users/", identifier))`

**Comment by jolheiser** [comment]
> I have missed that, thanks. But i think `trimGiteaUrl()` should be not used anymore since it can cause problems like #35, maybe simply use `urlJoin()`

In general I agree, I was just trying to match code style.
Would you like me to clean up the other places for urlJoin as well? Or another PR?

**Comment by thomiceli** [comment]
If you don't mind then yes, it's fine in this PR. I also noticed we should use `urlJoin()` there too : https://github.com/thomiceli/opengist/blob/3366cde38574eca30aa90c5534df3b7ade649714/internal/web/auth.go#L206-L209

**Comment by jolheiser** [comment]
I've done it in both spots. Up to you if you want to also use `urlJoin` for the GitHub URLs, but since those are hard-coded I think it's fine either way.

**Pull request merged**